snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedRf2ExportRestService.java

```diff
@@ -0,0 +1,111 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import com.b2international.snowowl.core.ApplicationContext;
+import com.b2international.snowowl.core.attachments.AttachmentRegistry;
+import com.b2international.snowowl.core.attachments.InternalAttachmentRegistry;
+import com.b2international.snowowl.core.domain.ExportResult;
+import com.b2international.snowowl.snomed.core.domain.Rf2RefSetExportLayout;
+import com.b2international.snowowl.snomed.core.rest.domain.SnomedRf2ExportConfiguration;
+import com.b2international.snowowl.snomed.datastore.config.SnomedCoreConfiguration;
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * @since 7.5
+ */
+@Api(value = "Export", description="Export", tags = "export")
+@Controller
+@RequestMapping(value="/{path:**}/export")
+public class SnomedRf2ExportRestService extends AbstractSnomedRestService {
+
+	@Autowired
+	private AttachmentRegistry attachments;
+	
+	@ApiOperation(
+		value="Export SNOMED CT content to RF2", 
+		notes="Exports SNOMED CT content from the given branch to RF2."
+	)
+	@ApiResponses({
+		@ApiResponse(code=200, message="OK")
+	})
+	@GetMapping
+	public @ResponseBody ResponseEntity<?> export(
+			@ApiParam(value = "The branch path", required = true)
+			@PathVariable(value="path")
+			final String branch,
+
+			final SnomedRf2ExportConfiguration params,
+			
+			@ApiParam(value = "Accepted language tags, in order of preference")
+			@RequestHeader(value=HttpHeaders.ACCEPT_LANGUAGE, defaultValue="en-US;q=0.8,en-GB;q=0.6", required=false) 
+			final String acceptLanguage) {
+		
+		final Rf2RefSetExportLayout globalExportLayout = ApplicationContext.getServiceForClass(SnomedCoreConfiguration.class).getExport().getRefSetExportLayout();
+		
+		final ExportResult exportedFile = SnomedRequests.rf2().prepareExport()
+			.setReleaseType(params.getType())
+			.setExtensionOnly(params.isExtensionOnly())
+			.setLocales(acceptLanguage)
+			.setIncludePreReleaseContent(params.isIncludeUnpublished())
+			.setModules(params.getModuleIds())
+			.setRefSets(params.getRefSetIds())
+			.setCountryNamespaceElement(params.getNamespaceId())
+			// .setNamespaceFilter(namespaceFilter) is not supported on REST, yet
+			.setTransientEffectiveTime(params.getTransientEffectiveTime())
+			.setStartEffectiveTime(params.getStartEffectiveTime())
+			.setEndEffectiveTime(params.getEndEffectiveTime())
+			.setRefSetExportLayout(params.getRefSetLayout() == null ? globalExportLayout : params.getRefSetLayout())
+			.build(this.repositoryId, branch)
+			.execute(getBus())
+			.getSync();
```

Comment:
Can we wait for the results here? What is the currently configured timeout value?

Reply:
No timeout value at the moment, we wait indefinitely.
We can best guess a proper timeout value based on the selected export configuration, but not sure if that's a good idea. Not sure if we can convert this to an async method either, my guess is that Spring needs the HTTP thread to finish the file download work.